recipe/build.sh

```diff
@@ -1,12 +1,24 @@
 #!/bin/bash
 
-cp -r ${BUILD_PREFIX}/share/libtool/build-aux/config.* .
+# Get an updated config.sub and config.guess
+cp $BUILD_PREFIX/share/gnuconfig/config.* . || true
 
-export LD_LIBRARY_PATH=$PREFIX/lib:$LD_LIBRARY_PATH
+if [[ "$target_platform" == "win-64" ]]; then
+  export PREFIX=${PREFIX}/Library
+fi
 
 ./configure --prefix=$PREFIX \
             --with-gmp=$PREFIX \
-            --enable-static
-make
-make check
+            --disable-static \
+            --enable-thread-safe
+
+make -j${CPU_COUNT}
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" != 1 && "${CROSSCOMPILING_EMULATOR}" == "" ]]; then
+  make check
+fi
 make install
+
+if [[ "$target_platform" == "win-64" ]]; then
+  cp ${PREFIX}/bin/libmpfr-6.dll ${PREFIX}/bin/mpfr.dll
```

Comment:
I worried by all this to be frank. We are changing the compiler stack and trying to make it so like it'll be a transparent process. I'm worried that doing this change will make us think that it's fine, and then we'll find out that the ABI is not compatible or whatever.

Could you (or @ifitchet , @skupr-anaconda) provide some more information/links about what we are trying to do, a short summary of the change and the potential impact and maybe a proof that we won't be breaking the dependents. Would this be possible?

Reply:
I have no reason to believe that MSYS2 themselves have changed anything but we are making a 7.5 year jump, so who knows.

In this particular instance, Dan changed the `bld.bat` to call `build.sh` which calls `configure` which generates different DLL names with attendant issues.

Reply:
>I have no reason to believe that MSYS2 themselves have changed anything

We were not using any msys2 stuff before. We were straight up compiling with MSVC before. Now e are building with a different compiler stack.

Reply:
Do we want to change to compiling with msys2? I can revert this change. But I think the issue will need to be solved later so msys2 compilers don't mean compiling with msys2 stack for all downstreams.
I wonder why conda-forge are using m2 rather than msvc here.

Reply:
@danpetry We ahve to be careful to not mix things here. msys2 provides multiple runtimes. Here we use msys2 with the MSVC runtime (I think, but again I asked that exact question jsut above and nobody is answering). There are differences between a runtime and a compiler.

Reply:
sure, but the msys2 stack produces an msys2-compatible import library, and you're talking about potential abi incompatibility

Reply:
that's how I mean dependents might need to be compiled with msys2 tools as well